src/controller/repository.ts

```diff
@@ -90,28 +116,32 @@ export const fetchFileList = catchAsync(
 		res: Response,
 		next: NextFunction
 	) => {
+		const { url, branch } = req.body;
+		const repoDir = dirName(url);
+		const repoPath = `${appsDirectory}/${repoDir}`;
+
 		try {
-			await deleteRepoFolderIfExist(appsDirectory, req.body.url);
-		} catch (err) {
-			next(
-				new AppError(
-					'error occurred in deleting repository directory',
-					500
-				)
+			// Delete existing repo folder if it exists
+			await deleteRepoFolderIfExist(appsDirectory, url);
+
+			// Clone the repository
+			await exec(`git clone ${url} ${repoPath} --depth=1 --no-checkout`);
+
+			// List files in the specified branch
+			const { stdout } = await exec(
+				`cd ${repoPath} && git ls-tree -r ${branch} --name-only`
 			);
-		}
-		await exec(
-			`cd ${appsDirectory} ; git clone ${req.body.url} --depth=1 --no-checkout`
-		);
 
-		const dirPath = `${appsDirectory}/${dirName(req.body.url)}`;
+			const files = stdout.trim().split('\n').filter(Boolean);
 
-		const { stdout } = await exec(
-			`cd ${dirPath} ; git ls-tree -r ${req.body.branch} --name-only; cd .. ; rm -r ${dirPath}`
-		);
+			// Clean up the cloned repository
+			await exec(`rm -r ${repoPath}`);
```

Comment:
Please do this with standard nodejs API, the command is not cross platform.

Reply:
Use this or it's async version:
https://stackoverflow.com/a/57866165

Reply:
Fixed 
https://github.com/metacall/faas/pull/59/commits/780a47a5886b197fcdfb40097fbb00c19b2d323e